code/modules/research/rdmachines.dm

```diff
@@ -39,9 +40,62 @@
 
 /obj/machinery/r_n_d/attack_hand(mob/user)
 	if(shocked)
-		shock(user,50)
+		if(shock(user,50))
+			return
 	if(panel_open)
 		wires.interact(user)
 
 
 
+/obj/machinery/r_n_d/attackby(obj/item/O, mob/user, params)
+	if (shocked)
+		if(shock(user,50))
+			return 1
+	if (default_deconstruction_screwdriver(user, "[initial(icon_state)]_t", initial(icon_state), O))
+		if(linked_console)
+			linked_console.linked_lathe = null
+			linked_console = null
+		return
+	if(exchange_parts(user, O))
+		return
+	if(default_deconstruction_crowbar(O))
+		return
+	if(Insert_Item(O, user))
+		return 1
+	else
+		return ..()
+
+//proc used to handle inserting items or reagents into r_n_d machines
+/obj/machinery/r_n_d/proc/Insert_Item(obj/item/I, mob/user)
+	return

```

Comment:
Introducing this proc helps us remove copypasta in the attackbys of the research machines.